packages/jest-leak-detector/src/index.js

```diff
@@ -35,6 +34,17 @@ export default class {
       );
     }
 
+    let weak;
+
+    try {
+      weak = require('weak');
+    } catch (err) {
+      throw new Error(
```

Comment:
This hides all sorts of errors. can we do an explicit check for `MODULE_NOT_FOUND`, and if not, rethrow original error?

(can use https://www.npmjs.com/package/optional and `rethrow`, but rolling our own should not be an issue)

Reply:
for convenience, `optional` would be something like

```js
const weak = optional('weak', { rethrow: true });

if (!weak) {
  throw new Error(
    'The leaking detection mechanism requires the "weak" package to work. ' +
      'Please make sure that you can install the native dependency on your platform.',
  );
}
```

current usage is just an `if` away, so might not be worth the extra dep 🙂

Reply:
I like the stricter check, let me add that into a new PR but as our own custom check; checking `error.code` should not require a new dependency 🙂

Reply:
Fixed in #5010.